velox/functions/prestosql/benchmarks/MapInputBenchmark.cpp

```diff
@@ -138,7 +148,46 @@ class NestedMapSumValuesAndKeysVector : public exec::VectorFunction {
   }
 };
 
-// Simple function implementation
+// This is an intresting point of comparison, here we use the vector reader
+// inside of the vector function. This allows to better isolate hidden factors
+// and to time different components.
+class NestedMapSumValuesAndKeysVectorUsingMapView
+    : public exec::VectorFunction {
+ public:
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& /* outputType */,
+      exec::EvalCtx* context,
+      VectorPtr* result) const override {
+    auto arg = args.at(0);
+
+    DecodedVector decoded_;
+    using exec_in_t = typename VectorExec::template resolver<
+        Map<int64_t, Map<int64_t, int64_t>>>::in_type;
+    decoded_.decode(*arg, rows);
+    VectorReader<Map<int64_t, Map<int64_t, int64_t>>> reader{decoded_};
+
+    // Prepare results
+    BaseVector::ensureWritable(rows, BIGINT(), context->pool(), result);
+    auto flatResult = (*result)->asFlatVector<int64_t>();
+
+    rows.applyToSelected([&](vector_size_t row) {
+      auto sum = 0;
+      for (const auto& entry : reader[row]) {
+        sum += entry.first;
+        for (const auto& entryInner : *entry.second) {
+          sum += entryInner.first;
+          sum += *entryInner.second;
+        }
+      }
+
+      flatResult->set(row, sum);
```

Comment:
`set()` sets both the value and null bit.

I see a common pattern where we ask for `mutableRawValues()` and then index directly into the raw values of the result to set the value, leaving the nulls in their initial state (presumably the initial state is not-null?). This is faster than `set()`, but it's not clear to me what the contract around nulls is when this pattern is used.

Reply:
in this benchmark I do not want to add any special optimizations that do not happen in the vector adapter.
basically i want to write the vector function in the way I expected that vector adapter would expand to focus on the mapView cost.

Reply:
or maybe i should make it faster and maybe fix the vector adapter if it does not do the thing here

Reply:
tried it, not needed.